test/backend-cpp/TestCategoryMapper.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---- TestCategoryMapper.cpp - Test CategoryMapper Implementation -----===//
+//
+// Copyright 2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file contains end-to-end tests for the CategoryMapper operator.
+//
+//===----------------------------------------------------------------------===//
+
+#include "test/backend-cpp/TestHelper.hpp"
+#include "llvm/Support/Debug.h"
+
+#define DEBUG_TYPE "backend-cpp"
+
+static const string SharedLibBase("./TestCategoryMapper_main_graph");
+
+namespace {
+
+class CategoryMapperTester {
+  using TestHelper = BackendCppTests::TestHelper;
+  TestHelper testHelper;
+
+public:
+  // Aggregate CategoryMapper attributes.
+  struct CMAttributes {
+    ArrayRef<int64_t> cat_int64s;
+    ArrayRef<StringRef> cat_strings;
+    int64_t default_int;
+    StringRef default_string;
+  };
+
+  CategoryMapperTester(MLIRContext &ctx) : testHelper(TestHelper(ctx)) {}
+
+  // Test CategoryMapper (with an input tensor of int64_t numbers).
+  bool testInt64ToStr(const CMAttributes &attributes, ArrayRef<int64_t> input,
+      ArrayRef<const char *> expectedOutput) {
+    assert(input.size() == expectedOutput.size() &&
+           "Expecting input/output to have the same size");
+
+    int64_t inputShape[] = {static_cast<int64_t>(input.size())};
+    auto inputType =
+        RankedTensorType::get(inputShape, testHelper.getBuilder().getI64Type());
+    auto outputType = RankedTensorType::get(
+        inputShape, onnxmlir::StringType::get(&testHelper.getContext()));
+
+    // Create the test code.
+    FuncOp funcOp = testHelper.createEmptyTestFunction(inputType, outputType);
+    createCategoryMapper(outputType, attributes, funcOp);
+    testHelper.createEntryPoint(funcOp);
+
+    // Compile the test.
+    if (!testHelper.compileTest(SharedLibBase)) {
+      llvm::errs() << "Failed to compile test case\n";
+      return false;
+    }
+
+    // Run the test and verify the result.
+    using OMTensorPtr = unique_ptr<OMTensor, decltype(&omTensorDestroy)>;
+
+    std::vector<OMTensorPtr> inputOMTs, expectedOutputOMTs;
+    auto inputOMT = OMTensorPtr(omTensorCreate((void *)input.data(), inputShape,
```

Comment:
I just noticed this - isn't the default to use c++ style casts?

Reply:
This being a C++ source file I would say yes! In general I agree that we should sue use a static_cast rather than a C style cast. Fixed.

Reply:
Great, thanks! I think there are a couple more remaining ones.

Reply:
Changed them.